moped-etl/arcgis/components_to_agol.py

```diff
@@ -23,6 +26,63 @@
 )
 
 
+def has_html_tags(html_string_to_check):
+    """Identify if a string contains HTML tags.
+
+    See: https://pypi.org/project/beautifulsoup4/
+
+    Args:
+        html_string_to_check (str): A string to test for HTML tags
+
+    Returns:
+        bool: True if the string contains HTML tags, False otherwise
+    """
+    return bool(BeautifulSoup(html_string_to_check, "html.parser").find())
+
+
+def is_valid_HTML_tag(html_string_to_check):
+    """Identify if a string contains valid HTML.
+
+    See: https://pypi.org/project/beautifulsoup4/
+
+    Args:
+        html_string_to_check (str): A string that contains HTML to test for valid HTML.
+
+    Returns:
+        bool: True if the string contains valid HTML, False otherwise
+    """
+    soup = BeautifulSoup(html_string_to_check, "html.parser")
+    return html_string_to_check == str(soup)
```

Comment:
this is definitely catching the case of the busted HTML, but i'm not totally understanding why. when i compare the before and after strings, all i see is that the `rel` and `href` tags are swapped. bs4 is obviously doing some linting, and it makes me wonder if this check is going to sweep up valid html from time to time.

i'm in favor of getting this patched up and running again asap, and maybe it'd be worth revisiting this there's no sense of urgency?

Reply:
@johnclary i should have include more info about this change. At the end of the status update, there is a break tag that is `<br>` and the str() method converts that to be self-closing `<br/>`. 

I can't explain the switch of the rel and href attributes. 🤷

Reply:
ah, thanks for walking me through that. i stared at this for a while and must have missed it. is there any hope that replacing the status update with the output of `str(soup)` makes it valid HTML that can be uploaded?

Reply:
john and i chatted a bit about this, and i really liked the idea but the AGOL API did not. 🙃 Sticking to text for now.